amazon-kinesis-client/src/test/java/software/amazon/kinesis/integration_tests/StreamExistenceManager.java

```diff
@@ -0,0 +1,101 @@
+package software.amazon.kinesis.integration_tests;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.http.Protocol;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryResponse;
+import software.amazon.awssdk.services.kinesis.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.kinesis.model.StreamStatus;
+import software.amazon.kinesis.config.KCLAppConfig;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Thread.sleep;
+
+@Data
+@Slf4j
+public class StreamExistenceManager {
+    private final KinesisAsyncClient client;
+    private final KCLAppConfig testConfig;
+
+    public StreamExistenceManager(KCLAppConfig config) throws URISyntaxException, IOException {
+        this.testConfig = config;
+        this.client = config.buildAsyncKinesisClient(Protocol.HTTP1_1);
```

Comment:
This ignores any protocol specified by the `KCLAppConfig`

Reply:
This client is for the DescribeStreams API call. For this, it does not matter which HTTP protocol is being used so a default is put here. When making PutRecord, GetRecord calls, the Consumer Protocol specified in KCLAppConfig is used. However, I will update this so that they are consistent.